src/utilities/array.ts

```diff
@@ -56,7 +63,22 @@ export const sortArray = (inputArray: Array<any>, sortOptions: Array<SortProps>)
 
 export const groupBy = function (inputArray, groupProp) {
     return inputArray.reduce((groups, item) => {
-        (groups[item[groupProp]] = groups[item[groupProp]] || []).push(item);
+        let result = item;
+        let splits = groupProp.split('.');
+        for (let i = 0; i < splits.length; i++) {
```

Comment:
won't this for loop just set the result to the deepest value?

Reply:
It will. Can it be done better?

Reply:
I saw a method written by @hrestakd `getObjectValue` so I will try to use that instead

Reply:
ah, sorry, my bad, now that i am looking at this again i see what you are actually doing here. All good.

Reply:
the method itself was really fast but the sorting for tree grid was the slow part so I removed it

Reply:
For QuickGrid purposes it has little to no performance impact